motion/thruster_interface_auv_cpp/src/thruster_interface_auv_ros.cpp

```diff
@@ -0,0 +1,77 @@
+    auto thruster_direction = this->get_parameter("propulsion.thrusters.thruster_direction").as_integer_array();
+    auto thruster_PWM_offset = this->get_parameter("propulsion.thrusters.thruster_PWM_offset").as_integer_array();
+    auto thruster_PWM_min = this->get_parameter("propulsion.thrusters.thruster_PWM_min").as_integer_array();
+    auto thruster_PWM_max = this->get_parameter("propulsion.thrusters.thruster_PWM_max").as_integer_array();
+    this->thrust_timer_period_ = 1.0 / this->get_parameter("propulsion.thrusters.thrust_update_rate").as_double();
+
+    //coeffs.yaml parameters
+    std::map<int, std::map<std::string, std::vector<double>>> coeffs;
+    std::vector<int> voltage_levels = {10, 12, 14, 16, 18, 20};
+    for (int voltage : voltage_levels) {
+        std::string left_param = "coeffs." + std::to_string(voltage) + "V.LEFT";
+        std::string right_param = "coeffs." + std::to_string(voltage) + "V.RIGHT";
+        this->declare_parameter<std::vector<double>>(left_param); //declare coeffs.10V.LEFT
+        this->declare_parameter<std::vector<double>>(right_param);
+
+        auto left_coeffs = this->get_parameter(left_param).as_double_array();
+        auto right_coeffs = this->get_parameter(right_param).as_double_array();
+
+        coeffs[voltage]["LEFT"] = left_coeffs; //save to coeffs[10]["LEFT"]
+        coeffs[voltage]["RIGHT"] = right_coeffs;
+    }
+
+    // Initialize thruster driver
+    this->thruster_driver_ = ThrusterInterfaceAUVDriver(
+        1, 0x21, 16.0,
+        std::vector<int>(thruster_mapping.begin(), thruster_mapping.end()),
+        std::vector<int>(thruster_direction.begin(), thruster_direction.end()),
+        std::vector<int>(thruster_PWM_offset.begin(), thruster_PWM_offset.end()),
+        std::vector<int>(thruster_PWM_min.begin(), thruster_PWM_min.end()),
+        std::vector<int>(thruster_PWM_max.begin(), thruster_PWM_max.end()),
+        coeffs
+    );
+
+    //Declare "thruster_forces_array" in case no topic comes in at the beginning
+    this->thruster_forces_array_ = std::vector<double>(8, 0.00);
+
+    timer_ = this->create_wall_timer(
+        std::chrono::duration<double>(this->thrust_timer_period_),
+        std::bind(&ThrusterInterfaceAUVNode::timer_callback, this)
+    );
+
+    RCLCPP_INFO(this->get_logger(), "\"thruster_interface_auv_cpp_node\" has been started");
+}
+
+void ThrusterInterfaceAUVNode::thruster_forces_callback(const vortex_msgs::msg::ThrusterForces::SharedPtr msg) {
+    this->thruster_forces_array_ = msg->thrust;
+}
+
+void ThrusterInterfaceAUVNode::timer_callback() {
+    std::vector<int16_t> thruster_pwm_array = thruster_driver_.drive_thrusters(this->thruster_forces_array_);
+
+    //publish PWM values for debugging
+    std_msgs::msg::Int16MultiArray pwm_message;
+    pwm_message.data = thruster_pwm_array;
+    thruster_pwm_publisher_->publish(pwm_message);
```

Comment:
Should have a ros param boolean flag that only publishes if the flag is set so we dont publish when not needed. The boolean flag can also conditionally initialize the publisher so we don't initialize it if not needed. It would be nice to be able to toggle this during runtime. You can have a look at https://github.com/vortexntnu/vortex-image-filtering/blob/main/image-filtering/src/image_filtering_ros.cpp for inspiration on how this can be done